### PR TITLE
Fixes #5334 - removes traces of Elastic Search

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

**Comment by beav** [comment]
There may still be a couple of traces in the backup/restore script and in the ping test.

LGTM aside from that though :koala:

**Comment by ehelms** [comment]
If this is designed to remove the last bits of ES, please update this to 'fixes' and also update the Redmine ticket to better describe what this is fixing.

**Comment by ehelms** [comment]
Looks like this is what @beav is referring to --https://github.com/Katello/katello/blob/master/spec/models/ping_spec.rb#L11-L12

Also, this can now be removed and any references to it (check application_controller.rb) -- https://github.com/Katello/katello/blob/master/config/katello.yaml.example#L16

This directory can be removed -- https://github.com/cfouant/katello/tree/166745889808a2279722209eee5d57b627894aaa/test/fixtures/vcr_cassettes/elasticsearch

File: app/lib/actions/katello/provider/reindex_subscriptions.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module Actions
-  module ElasticSearch
+  module Katello
     module Provider
-      class ReindexSubscriptions < ElasticSearch::Abstract
+      class ReindexSubscriptions < Actions::EntryAction
+        middleware.use Actions::Middleware::KeepCurrentUser
```
Comment (ehelms) [inline app/lib/actions/katello/provider/reindex_subscriptions.rb]: Blank line after this to separate logical chunks of code for readability.

File: app/lib/actions/katello/repository/filtered_index_content.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 module Actions
-  module ElasticSearch
+  module Katello
     module Repository
-      class FilteredIndexContent < ElasticSearch::Abstract
+      class FilteredIndexContent < Actions::EntryAction
+        middleware.use Actions::Middleware::KeepCurrentUser
```
Comment (ehelms) [inline app/lib/actions/katello/repository/filtered_index_content.rb]: Blank line after this to separate logical chunks of code for readability.

File: app/lib/actions/katello/repository/index_content.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 module Actions
-  module ElasticSearch
+  module Katello
     module Repository
-      class IndexContent < ElasticSearch::Abstract
+      class IndexContent < Actions::EntryAction
+        middleware.use Actions::Middleware::KeepCurrentUser
```
Comment (ehelms) [inline app/lib/actions/katello/repository/index_content.rb]: Blank line after this to separate logical chunks of code for readability.

File: app/models/katello/ext/indexed_model.rb [hunk 4]
```diff
@@ -11,10 +11,6 @@ def self.display_attributes
         end
 
         if Rails.env.development? || Rails.env.production?
-          include Tire::Model::Search
```
Comment (ehelms) [inline app/models/katello/ext/indexed_model.rb]: I think this class, IndexedModel, can be removed all together as well as all references to it.

**Comment by ehelms** [comment]
Parts of (or perhaps the entire file) can be removed -- https://github.com/cfouant/katello/blob/166745889808a2279722209eee5d57b627894aaa/spec/helpers/search_helper_methods.rb

**Comment by johnpmitsch** [comment]
Looks like [here](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/host_collections_controller.rb#L95) and [here] (https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/host_collections_controller.rb#L117) can be removed

**Comment by cfouant** [comment]
@johnpmitsch - should all traces of System.index be removed? I note at least one other.

**Comment by ehelms** [comment]
Can you also update the PR message to say fixes?

**Comment by johnpmitsch** [comment]
@cfouant I think so, I assume you mean https://github.com/Katello/katello/blob/master/lib/katello/tasks/clean_backend_objects.rake#L19 --  maybe @ehelms can confirm that line is ok to remove?

**Comment by ehelms** [comment]
@cfouant Yea, instances of System.index can be removed. In general, <Model>.index was an elasticsearch concept.

**Comment by ehelms** [comment]
Based on IRC conversation, you will want to remove the webmock removal commit -- that should also get the tests back to green.

**Comment by ehelms** [comment]
ACK for me -- as @jlsherrill did a lot of the original ES work, I'd like him to do final ACK on this

**Comment by jlsherrill** [comment]
ACK from me :+1:   So long ES, we hardly knew ye  :ship:

**Comment by mccun934** [comment]
I know this is closed but I wanted to comment with a :+1: and :clap:

**Pull request merged**